agent/fw_drupal8.c

```diff
@@ -605,6 +605,94 @@ NR_PHP_WRAPPER(nr_drupal94_invoke_all_with_clean) {
 NR_PHP_WRAPPER_END
 #endif  // OAPI
 
+static bool nr_validate_key_val_arr(nr_php_string_hash_key_t* key, zval* val) {
+  if (NULL == key || 0 == ZEND_STRING_LEN(key)
+      || 0 == nr_php_is_zval_valid_array(val)
+      || 0 == zend_hash_num_elements(Z_ARRVAL_P(val))) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+/*
+ * Purpose: Instrument Drupal Attribute Hooks for Drupal 11.1+
+ *
+ * Params: 1. A zval pointer to the moduleHandler instance in use by Drupal.
+ *
+ * Return: bool
+ *
+ */
+static bool nr_drupal_hook_attribute_instrument(zval* module_handler) {
+  zval* hook_implementation_map = NULL;
+
+  nr_php_string_hash_key_t* hook_key = NULL;
+  zval* hook_val = NULL;
+  nr_php_string_hash_key_t* class_key = NULL;
+  zval* class_val = NULL;
+  nr_php_string_hash_key_t* method_key = NULL;
+  zval* module_val = NULL;
+
+  char* hookpath = NULL;
+
+  hook_implementation_map = nr_php_get_zval_object_property(
+      module_handler, "hookImplementationsMap");
+
+  if (!nr_php_is_zval_valid_array(hook_implementation_map)) {
+    nrl_warning(NRL_FRAMEWORK,
+                "hookImplementationsMap property not a valid array");
```

Comment:
How frequently will this message be logged with agent's default log level when running Drupal < 11.1?

Reply:
Does this need to be warning vs something like verbosedebug?

Reply:
fda25fe30b931b180adb4472eb8363aab24e47fb